File: app/src/main/java/de/westnordost/streetcomplete/quests/recycling/AddRecyclingTypeForm.java
package de.westnordost.streetcomplete.quests.recycling;

import de.westnordost.streetcomplete.R;
import de.westnordost.streetcomplete.quests.ImageListQuestAnswerFragment;
import de.westnordost.streetcomplete.view.Item;

public class AddRecyclingTypeForm extends ImageListQuestAnswerFragment
{
	private final Item[] TYPES = new Item[] {
			new Item("overground", R.drawable.recycling_container, R.string.overground_recycling_container),
			new Item("underground", R.drawable.recycling_container_underground, R.string.underground_recycling_container),
			new Item("centre", R.drawable.recycling_centre, R.string.recycling_centre)
	};

	@Override protected Item[] getItems() { return TYPES; }
	@Override protected int getItemsPerRow() { return 3; }
}

File: app/src/main/java/de/westnordost/streetcomplete/quests/recycling/AddRecyclingTypeForm.kt
package de.westnordost.streetcomplete.quests.recycling

import de.westnordost.streetcomplete.R
import de.westnordost.streetcomplete.quests.AImageListQuestForm
import de.westnordost.streetcomplete.quests.recycling.RecyclingType.OVERGROUND_CONTAINER
import de.westnordost.streetcomplete.quests.recycling.RecyclingType.RECYCLING_CENTRE
import de.westnordost.streetcomplete.quests.recycling.RecyclingType.UNDERGROUND_CONTAINER
import de.westnordost.streetcomplete.view.image_select.Item

class AddRecyclingTypeForm : AImageListQuestForm<RecyclingType, RecyclingType>() {

    override val items = listOf(
        Item(OVERGROUND_CONTAINER, R.drawable.recycling_container, R.string.overground_recycling_container),
        Item(UNDERGROUND_CONTAINER, R.drawable.recycling_container_underground, R.string.underground_recycling_container),
        Item(RECYCLING_CENTRE, R.drawable.recycling_centre, R.string.recycling_centre)
    )

    override val itemsPerRow = 3

    override fun onClickOk(selectedItems: List<RecyclingType>) {
        applyAnswer(selectedItems.single())
    }
}
